Workflow publish validate

chent00ng opened the pull request.

chent00ng committed:
Mark workflow publish and validate as in development - assigned to c6:c4:e5:e8:c6:4b

chent00ng committed:
Implement workflow publish and validate functionality

- Add new WorkflowRunService with validation, publishing, and execution
- Implement concurrent validation for workflow nodes to improve performance
- Add api/workflow/run endpoint in WorkflowController
- Support comprehensive error handling with UserFriendlyException
- Validate user-configured parameters in workflow view nodes
- Simplify publish logic using publishedAgent.Id directly
- Optimize code structure and remove redundant complexity

chent00ng committed:
add WorkflowId return

chent00ng committed:
Fix workflow execution timing issue - implement event availability check with retry logic

chent00ng committed:
Add 500ms retry delay between workflow event availability checks for better success rate

chent00ng committed:
Remove Chinese comment for better code readability

chent00ng committed:
Optimize workflow event checking with direct agent access and improved code formatting

chent00ng committed:
Fix workflow agent ID issue - extract correct WorkflowCoordinatorAgent ID instead of ViewAgent ID

chent00ng committed:
Enhance PublishWorkflowAsync with detailed logging for better problem diagnosis

chent00ng committed:
Enhance workflow event detection - add EventInfoList logging and increase retry attempts to 20 with 1-second intervals

chent00ng committed:
skip validate test

chent00ng committed:
Refactor workflow validation logic to properly validate each node properties like PublishWorkflowAsync approach

chent00ng committed:
Optimize workflow validation with concurrent processing for better performance

chent00ng committed:
skip validate

chent00ng committed:
Add standardized [AgentValidation] prefix to all AgentValidationService logs for better filtering and debugging

chent00ng committed:
update private method input

chent00ng committed:
Migrate agent validation logic from AgentValidationService to WorkflowRunService - make WorkflowRunService self-contained

chent00ng committed:
skip validate

chent00ng committed:
Refactor WorkflowRunService validation architecture for consistency and simplification

- Aligned validation logic with AgentService.SetupConfigurationData pattern
- Added IGAgentFactory dependency and removed redundant FindConfigTypeByAgentNamespace methods
- Use random GUID instead of fixed string for temporary grain IDs to avoid conflicts
- Ensured data flow consistency with WorkflowViewService (JsonProperties -> Dictionary -> JSON)
- Simplified validation logic: removed data annotation validation, simplified schema error messages
- Changed error handling to throw UserFriendlyException directly instead of returning ConfigValidationResultDto
- Maintained comprehensive logging with [AgentValidation] prefixes for debugging

chent00ng committed:
skip schema validate

chent00ng committed:
skip ValidatableObject

chent00ng committed:
update validate method

chent00ng committed:
skip ValidateWorkflowNodePropertiesAsync

chent00ng committed:
skip validate

chent00ng committed:
add ValidateWorkflowConfigurationAsync

chent00ng committed:
Fix RouterGAgent unit test JSON parsing error

- Enhanced MockKernelBuilderFactory to return valid RouterOutputSchema JSON responses
- Added intelligent response logic for router prompts detection
- Added System.Linq using statement for LINQ operations
- Fixed project references formatting consistency
- Added Newtonsoft.Json package reference for JSON handling

This resolves the RouterGAgent.RouteNext JSON deserialization failure at line 139.

chent00ng committed:
Merge branch 'dev' into feature/workflow-publish-validate

chent00ng committed:
添加WorkflowRunService单元测试

- 创建WorkflowRunServiceTests.cs，包含13个测试用例
- 测试覆盖RunWorkflowAsync方法的各种场景
- 包括参数验证、错误处理、依赖服务调用验证
- 所有测试用例均通过，测试覆盖率良好
- 参考WorkflowViewServiceTests.cs的编写风格和模式

chent00ng committed:
Add comprehensive WorkflowController unit tests

- Created WorkflowControllerTests.cs with 20 test cases covering all 3 API endpoints
- Tests include: GenerateAsync, GenerateTextCompletionAsync, RunWorkflowAsync
- Improved test coverage from 0.00% to 100% for WorkflowController
- Fixed duplicate GenerateWorkflowRequestDto class definition issue
- All tests passing with comprehensive validation of HTTP attributes, error handling, and service integration
- Enhanced overall application test coverage and code quality

chent00ng committed:
Merge branch 'dev' into feature/workflow-publish-validate

chent00ng committed:
Enhance WorkflowRunService test coverage with additional validation test cases

- Added 3 new test cases for better edge case coverage
- Tests include: null JSON properties, empty JSON properties validation
- Improved overall WorkflowRunService test reliability and coverage
- All 23 WorkflowRunService tests now pass successfully
- Enhanced test coverage from 76.32% towards 80% target

chent00ng committed:
Clean up WorkflowRunService code - remove commented code and debug comments

- Removed commented out IGAgentManager dependencies
- Cleaned up unnecessary debug comments and step annotations
- Improved code readability while maintaining functionality
- No functional changes, only code cleanup

chent00ng committed:
Merge branch 'dev' into feature/workflow-publish-validate

arg-foo commented on station/src/Aevatar.Application.Contracts/AgentValidation/ValidationRequestDto.cs:
```diff
@@ -7,15 +7,6 @@ namespace Aevatar.AgentValidation;
-    /// Required: Complete GAgent Namespace (e.g., "Aevatar.GAgents.DatabaseGAgent")
-    /// </summary>
-    [Required(ErrorMessage = "GAgent Namespace is required")]
     public string GAgentNamespace { get; set; } = string.Empty;
```
Should this have Description?

chent00ng replied:
I forgot to restore it, please wait.

arg-foo commented on station/src/Aevatar.HttpApi/Controllers/WorkflowController.cs:
```diff
@@ -12,21 +11,21 @@ namespace Aevatar.Controllers
     [ApiController]
     [Route("api/workflow")]
-    [Authorize]
+    // [Authorize]
```
Why is this commented?

chent00ng replied:
Forgot to uncomment it, I'll restore it

arg-foo commented on station/src/Aevatar.Application/Service/WorkflowRunService.cs:
```diff
@@ -0,0 +1,345 @@
+        };
+    }
+    
+    private async Task ValidateWorkflowConfigurationAsync(Guid viewAgentId)
```
How does this inform the upper function that it failed? Through exceptions?

chent00ng replied:
Yes, the verification results are directly returned to the front-end via user-friendly methods.

arg-foo commented on station/src/Aevatar.Application/Service/WorkflowRunService.cs:
```diff
@@ -0,0 +1,345 @@
+        }
+        
+        var nodeAgentProperties = JsonConvert.DeserializeObject<Dictionary<string, object>>(workflowNode.JsonProperties);
+        var initializationParam = JsonConvert.SerializeObject(nodeAgentProperties);
```
Why deserialize then serialize immediately? Is it to check for schema issues? If so, exception should be thrown and caught within the function

chent00ng committed:
Merge dev branch with conflict resolution - resolved conflicts in project_tracker.md, Application.csproj, and WorkflowRunService.cs

chent00ng committed:
Fix remaining merge conflict markers in WorkflowRunService.cs - resolve EventProperties assignment conflict

chent00ng committed:
Fix CS0854 error: Add missing optional parameter to GetTypeSchema mock setup

chent00ng committed:
Merge branch 'dev' into feature/workflow-publish-validate

chent00ng committed:
Fix PropertyJsonSchema null issue in AgentService with comprehensive error handling and fallbacks

chent00ng committed:
Simplify workflow coordinator ID access using direct property instead of JSON parsing

- Add WorkflowCoordinatorGAgentId property to AgentDto
- Modify WorkflowRunService to use direct property access for coordinator ID
- Remove duplicate WorkflowRunDto.cs file that caused compilation conflicts
- Simplify WorkflowId assignment using publishedAgent.WorkflowCoordinatorGAgentId
- Remove complex JSON deserialization logic in favor of direct property access

chent00ng committed:
revert required

chent00ng committed:
Merge branch 'dev' into feature/workflow-publish-validate

chent00ng committed:
Add JSON exception handling to ValidateWorkflowNodePropertiesAsync method

- Add try-catch blocks for JsonConvert.DeserializeObject and JsonConvert.SerializeObject
- Handle JsonReaderException for invalid JSON format in JsonProperties
- Handle JsonSerializationException for deserialization and serialization failures
- Provide clear error messages with node name and exception details
- Improve robustness of workflow node properties validation

chent00ng committed:
update ns

arg-foo reviewed (dismissed):
LGTM

chent00ng committed:
fix ut ns

chent00ng committed:
Fix WorkflowRunRequestDto compilation error in WorkflowControllerTests

- Add missing using directive for Aevatar.WorkflowRun namespace
- Resolve CS0246 error: The type or namespace name 'WorkflowRunRequestDto' could not be found
- Ensure proper namespace imports for workflow run related DTOs in test files

chent00ng committed:
Fix WorkflowRunServiceTests failures with proper null handling and validation

- Add null check for publishedAgent.WorkflowCoordinatorGAgentId with UserFriendlyException
- Add validation for publishedAgent.Properties null check
- Add JSON deserialization validation with proper error handling
- Update test mock data to properly set WorkflowCoordinatorGAgentId in AgentDto
- Fix mock data for error test cases to explicitly set null values
- Use Newtonsoft.Json.JsonException to resolve ambiguous reference issue
- All 23 WorkflowRunServiceTests now pass successfully
- Maintain proper error message expectations for test cases

chent00ng committed:
Merge branch 'dev' into feature/workflow-publish-validate

arg-foo reviewed (approved):
LGTM
